securedrop/crypto_util.py

```diff
@@ -12,51 +12,206 @@
+                 adjectives_file,
+                 gpg_key_dir):
+        self.__securedrop_root = securedrop_root
+        self.__word_list = word_list
+
+        if os.environ.get('SECUREDROP_ENV') == 'test':
+            # Optimize crypto to speed up tests (at the expense of security
+            # DO NOT use these settings in production)
+            self.__gpg_key_length = 1024
+            self.scrypt_params = dict(N=2**1, r=1, p=1)
+        else:  # pragma: no cover
+            self.__gpg_key_length = 4096
+            self.scrypt_params = scrypt_params
+
+        self.scrypt_id_pepper = scrypt_id_pepper
+        self.scrypt_gpg_pepper = scrypt_gpg_pepper
+
+        self.do_runtime_tests()
+
+        self.__gpg_key_dir = gpg_key_dir
+        self.gpg = gnupg.GPG(binary='gpg2', homedir=gpg_key_dir)
+
+        # map code for a given language to a localized wordlist
+        self.__language2words = {}  # type: Dict[Text, List[str]]
+
+        with open(nouns_file) as f:
+            self.nouns = f.read().splitlines()
+
+        with open(adjectives_file) as f:
+            self.adjectives = f.read().splitlines()
+
+    # Make sure these pass before the app can run
+    # TODO: Add more tests
+    def do_runtime_tests(self):
+        if self.scrypt_id_pepper == self.scrypt_gpg_pepper:
+            raise AssertionError('scrypt_id_pepper == scrypt_gpg_pepper')
+        # crash if we don't have srm:
+        try:
+            subprocess.check_call(['srm'], stdout=subprocess.PIPE)
+        except subprocess.CalledProcessError:
+            pass
+
+    def get_wordlist(self, locale):
+        # type: (Text) -> List[str]
+        """" Ensure the wordlist for the desired locale is read and available
+        in the words global variable. If there is no wordlist for the
+        desired local, fallback to the default english wordlist.
+
+        The localized wordlist are read from wordlists/{locale}.txt but
+        for backward compatibility purposes the english wordlist is read
+        from the config.WORD_LIST file.
+        """
+
+        if locale not in self.__language2words:
+            if locale != 'en':
+                path = os.path.join(self.__securedrop_root,
+                                    'wordlists',
+                                    locale + '.txt')
+                if os.path.exists(path):
+                    wordlist_path = path
+                else:
+                    wordlist_path = self.__word_list
+            else:
+                wordlist_path = self.__word_list
+
+            with open(wordlist_path) as f:
+                content = f.read().rstrip('\n').split('\n')
+                self.__language2words[locale] = content
+
+        return self.__language2words[locale]
+
+    def genrandomid(self,
+                    words_in_random_id=None,
+                    locale='en'):
+        if words_in_random_id is None:
+            words_in_random_id = self.DEFAULT_WORDS_IN_RANDOM_ID
```

Comment:
Seems like setting `self.DEFAULT_WORDS_IN_RANDOM_ID` as the default kwarg for `words_in_random_id` would obviate the need for these two lines of code (beginning `if words_in_random_id is None:`)

Reply:
It breaks because `self` isn't defined at class definition time. I did this so we could leave the `DEFAULT_WORDS_IN_RANDOM_ID` attached to the class, but if you don't like this pattern I can make this constant part of the module and then set it as the default in the function.

Reply:
```
>>> class Foo():
...     X = 1
...     def bar(self, baz=self.X):
...         return baz
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 3, in Foo
NameError: name 'self' is not defined
```

Reply:
ahhhh right, that's cool then